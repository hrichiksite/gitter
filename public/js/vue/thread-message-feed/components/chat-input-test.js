jest.mock('../../../utils/is-touch');

const mount = require('../../__test__/vuex-mount');
const { default: ChatInput } = require('./chat-input.vue');
import isTouch from '../../../utils/is-touch';

describe('thread-message-feed chat-input', () => {
  const defaultProps = {
    user: { displayName: 'John Smith' },
    thread: true
  };

  beforeEach(() => {
    isTouch.mockReturnValue(false);
  });

  it('matches snapshot for thread', () => {
    const { wrapper } = mount(ChatInput, defaultProps);
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('sending message', () => {
    it('should update draft message when user adds input', () => {
      const { wrapper, stubbedActions } = mount(ChatInput, defaultProps);
      wrapper.find({ ref: 'chatInputTextArea' }).element.value = 'hello';
      wrapper.find({ ref: 'chatInputTextArea' }).trigger('input');
      expect(stubbedActions.threadMessageFeed.updateDraftMessage).toHaveBeenCalledWith(
        expect.anything(),
        'hello',
        undefined
      );
    });

    it('should render draft message when it is in state', () => {
      const { wrapper } = mount(
        ChatInput,
        defaultProps,
        store => (store.state.threadMessageFeed.draftMessage = 'new message')
      );
      expect(wrapper.find({ ref: 'chatInputTextArea' }).element.value).toEqual('new message');
    });

    it('should trigger send action when enter key is pressed', () => {
      const { wrapper, stubbedActions } = mount(ChatInput, defaultProps, store => {
        store.state.user = { id: 'userId' };
      });
      wrapper.find({ ref: 'chatInputTextArea' }).trigger('keydown.enter');
      expect(stubbedActions.threadMessageFeed.sendMessage).toHaveBeenCalled();
    });
  });

  it('should edit last message when key up is pressed', () => {
    const { wrapper, stubbedActions } = mount(ChatInput, defaultProps, store => {
      store.state.user = { id: 'userId' };
    });
    wrapper.find({ ref: 'chatInputTextArea' }).trigger('keyup.up');
    expect(stubbedActions.threadMessageFeed.editLastMessage).toHaveBeenCalled();
  });
});
