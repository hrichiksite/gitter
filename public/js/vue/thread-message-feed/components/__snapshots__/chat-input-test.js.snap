// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thread-message-feed chat-input matches snapshot for thread 1`] = `
<footer
  class="chat-input thread"
  id="chat-input-wrapper"
>
  <div
    class="chat-input__container"
  >
    <div
      class="chat-input__area"
    >
      <div
        class="chat-input__avatar"
      >
        <avatar-stub
          size="s"
          user="[object Object]"
        />
      </div>
       
      <div
        class="chat-input-box__region"
        id="chat-input-box-region"
      >
        <form
          class="chat-input__box"
          name="chat"
        >
          <textarea
            autocomplete="off"
            autocorrect="off"
            autofocus="autofocus"
            class="chat-input__text-area"
            maxlength="4096"
            name="chat"
            placeholder="Click here to type a chat message."
            style="height: 0px;"
          />
        </form>
      </div>
    </div>
  </div>
</footer>
`;
